src/dma/mod.rs

```diff
@@ -41,28 +41,12 @@ use traits::{
 };
 
 /// Errors.
-#[derive(PartialEq)]
-pub enum DMAError<T> {
+#[derive(PartialEq, Debug, Copy, Clone)]
+pub enum DMAError {
     /// DMA not ready to change buffers.
-    NotReady(T),
+    NotReady,
```

Comment:
Instead of adding a `NotReady`, can we use an `nb::Result::WouldBlock`? I think it's essentially being used in the same way - a user needs to wait for the transfer to finish first.

Reply:
I think that's useful. But this error should also carry the unused buffer (otherwise that's lost). The alternative would be to have another closure that gets the unused buffer in case of error, making `transfer_next_with()` somewhat like `Result::map_or_else()`.

Reply:
That's a good point, but this PR doesn't support returning the unused buffer on failure anyways

Reply:
That regression is on the TODO list above.